ui/raidboss/data/05-shb/trial/varis-ex.js

```diff
@@ -0,0 +1,124 @@
+'use strict';
+[{
+  zoneRegex: {
+    en: /^Memoria Misera \(Extreme\)$/,
+  },
+  timelineFile: 'varis-ex.txt',
+  triggers: [
+    {
+      id: 'VarisEx Phase 2',
+      regex: Regexes.startsUsing({ source: 'Varis Yae Galvus', id: '4CCC', capture: false }),
+      run: function(data) {
+        data.phase = 'phase2';
+      },
+    },
+    {
+      id: 'VarisEx Ignis Est',
+      regex: Regexes.startsUsing({ source: 'Ignis Est', id: '4CB6', capture: false }),
+      delaySeconds: 2,
+      response: Responses.getOut(),
+    },
+    {
+      id: 'VarisEx Ventus Est',
+      regex: Regexes.startsUsing({ source: 'Ventus Est', id: '4CC7', capture: false }),
+      delaySeconds: 2,
+      response: Responses.getIn('info'),
+    },
+    {
+      id: 'VarisEx Altius',
+      regex: Regexes.startsUsing({ source: 'Varis Yae Galvus', id: '4CCA', capture: false }),
+      infoText: {
+        en: 'Bait Slashes',
+      },
+    },
+    {
+      id: 'VarisEx Citius',
+      regex: Regexes.startsUsing({ source: 'Varis Yae Galvus', id: '4CF0' }),
+      alertText: function(data, matches) {
+        if (data.me == matches.target) {
+          return {
+            en: 'Buster on YOU',
+          };
+        }
+        if (data.role == 'dps') {
+          return {
+            en: 'Avoid Tanks',
+          };
+        }
+        return {
+          en: 'Buster on ' + data.ShortName(matches.target),
+        };
+      },
+    },
+    {
+      id: 'VarisEx Alea lacta Est',
+      regex: Regexes.startsUsing({ source: 'Varis Yae Galvus', id: '4CD2', capture: false }),
+      response: Responses.getBehind('alarm'),
+    },
+    {
+      id: 'VarisEx Alea lacta Est Front',
+      regex: Regexes.ability({ source: 'Varis Yae Galvus', id: '4CD5', capture: false }),
+      alertText: {
+        en: 'Go Front',
+      },
+    },
+    {
+      id: 'VarisEx Electrified Gunshield',
+      regex: Regexes.ability({ source: 'Varis Yae Galvus', id: '4CD7', capture: false }),
+      delaySeconds: 22,
```

Comment:
What's the exact timing between 4CD7 and the knockback in the log? If possible, I want to give exactly a 5 second warning.  #1322 has 20 seconds, so I'm just curious.

Reply:
I've used these triggers and I believe a 20 second warning would have you use your kb mitigation too early.

Reply:
Note also that at least one knockback is paired with a Chariot mechanic (Ignis Est.) It is probably important to make sure that the knockback warning is displayed after `getOut()` in that situation. (Speaking from experience playing Bard in Stormblood, not everyone uses anti-knockback abilities consistently.)

Reply:
I assume it is 26.6 seconds, why I set the `delaySeconds` as 20 sec just I played Black Mage, so I need an earlier notice so that I can plan which GCD can I insert my anti-knockback. (Although most of the time I have a `Fire 3 proc` or `Xenoglossy` allowed me to "Surecast")
I think 5 sec was still fine. But I want to announce player with timeline text too.